src/main/java/org/yaml/snakeyaml/constructor/SafeConstructor.java

```diff
@@ -114,6 +117,8 @@ protected void processDuplicateKeys(MappingNode node, boolean forceStringKeys) {
           if (!isAllowDuplicateKeys()) {
             throw new DuplicateKeyException(node.getStartMark(), key,
                 tuple.getKeyNode().getStartMark());
+          } else if (isWarnOnDuplicateKeys() && log.isLoggable(Logger.Level.WARNING)) {
+            log.warn(String.format("duplicate keys found : {}", key));
```

Comment:
`%s`
`String.format` does not use slf4j way of formatting string ;) 
https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html

This is the problem in this PR. It actually does not check that expected string is logged :( 

I understand that it is actually very difficult to intercept log messages with the current logger usage in snakeyaml, but it would be nice if solution is found to really test it.

Reply:
fix this one, and I will then fix the test - I have a "solution" to test logs :)

Reply:
> `%s` `String.format` does not use slf4j way of formatting string ;) https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html
> 
> This is the problem in this PR. It actually does not check that expected string is logged :(
> 
> I understand that it is actually very difficult to intercept log messages with the current logger usage in snakeyaml, but it would be nice if solution is found to really test it.

Thanks for your time and for the feedback.
You are totally right. It was my fault, I replaced the formatting string.
I think I also added an actual check of the logged message in the test case.
Let me know if the proposed solution is acceptable for you.